[FIX] mrp_partial_production: Missing quant reservation when producing t#11323

hugho-ad opened the pull request:
[FIX] mrp_partial_production: Missing quant reservation when producing
    less than the qty reserved
    
    Before this commit when you recorded production less than the qty you
    reserved on raw materials, the stock quants did not get reserved with
    the remaining qty from previous reservation but inventory lines.
    
    This causes a conflict when trying to cancel/unreserve the mrp
    production getting following error:
    
    It is not possible to unreserve more products of %s than you have in stock.
    
    After this commit, now quants reservation it's consistent with inventory
    reservation.

hugho-ad committed:
[FIX] mrp_partial_production: Missing quant reservation when producing
less than the qty reserved

Before this commit when you recorded production less than the qty you
reserved on raw materials, the stock quants did not get reserved with
the remaining qty from previous reservation but inventory lines.

This causes a conflict when trying to cancel/unreserve the mrp
production getting following error:

It is not possible to unreserve more products of %s than you have in stock.

After this commit, now quants reservation it's consistent with inventory
reservation.

luisg123v commented on mrp_partial_production/models/stock.py:
```diff
@@ -28,10 +29,14 @@ def _split(self, qty, restrict_partner_id=False):
-        self.browse(new_move_id)._recompute_state()
+        new_move = self.browse(new_move_id)
+        new_move._update_reserved_quantity(
+            new_move.product_uom_qty, available_quantity, new_move.location_id, strict=False)
```
Are you sure it should be strict=False?

hugho-ad replied:
yes, we got it from after debugging to following line
https://github.com/odoo/odoo/blob/11.0/addons/stock/models/stock_move.py#L950

luisg123v reviewed (approved):
LGTM, just provide a commit description, please.

hugho-ad committed:
[FIX] mrp_partial_production: Ensure reservation on strict=True

Since we are customizing split move in order to record partial
production, it's needed to ensure same quants when producing partial and
reserving inventory and materials.

hugho-ad commented:
@luisg123v 

as we talk, now we ensure reserving inventory materials on strict mode since we should keep the same inventory when producing less than the reserved and avoid re-reserve on another child locations

hugho-ad commented:
@keylor2906 @luisg123v  @JulioSerna 
please review

luisg123v reviewed (approved):
:+1:

CC @belisac

keylor2906 commented:
@hugho-ad Missing line at EOF, but LGTM

hugho-ad commented:
@keylor2906 Done!
thanks!

hugho-ad commented:
@JulioSerna 

Your comments are welcome

luisg123v commented:
This has been approved by the customer, so merging.

I'll squash because the 2nd commit is part of the reservation and it was a required change we noticed after the fix was implemented.

hugho-ad commented:
:+1:
